src/Microsoft.Framework.FileSystemGlobbing/Abstractions/DirectoryInfoWrapper.cs

```diff
@@ -8,16 +8,16 @@ namespace Microsoft.Framework.FileSystemGlobbing.Abstractions
 {
     public class DirectoryInfoWrapper : DirectoryInfoBase
     {
-        private DirectoryInfo DirectoryInfo;
+        private readonly DirectoryInfo _directoryInfo;
 
         public DirectoryInfoWrapper(DirectoryInfo directoryInfo)
         {
-            DirectoryInfo = directoryInfo;
+            _directoryInfo = directoryInfo;
         }
 
         public override IEnumerable<FileSystemInfoBase> EnumerateFileSystemInfos(string searchPattern, SearchOption searchOption)
         {
-            foreach (var fileSystemInfo in DirectoryInfo.EnumerateFileSystemInfos(searchPattern, searchOption))
+            foreach (var fileSystemInfo in _directoryInfo.EnumerateFileSystemInfos(searchPattern, searchOption))
```

Comment:
It's up to the implementation of `DirectoryInfoBase`. The consumer of `Matcher` decides what's the implementation is: https://github.com/aspnet/FileSystem/blob/filesystemglobbing.followup.1/src/Microsoft.Framework.FileSystemGlobbing/Matcher.cs#L27